### ISSUE TITLE
Task-2

### ISSUE THREAD

**Issue by sharob24** [vaishnavi1-lab/CodTech-Tasks#6]
from sklearn.datasets import load_diabetes
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Load dataset
data = load_diabetes()
X = pd.DataFrame(data.data, columns=data.feature_names)
y = data.target

# Split & Train
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
reg_model = LinearRegression()
reg_model.fit(X_train, y_train)

# Evaluate
y_pred = reg_model.predict(X_test)
print("MSE:", mean_squared_error(y_test, y_pred))

data = pd.read_csv('https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv')
data.head()
 
# Drop unnecessary columns
data.drop(['Name', 'Cabin', 'Ticket'], axis=1, inplace=True)

# Fill missing values
data['Age'].fillna(data['Age'].median(), inplace=True)
data['Embarked'].fillna(data['Embarked'].mode()[0], inplace=True)

# Convert categorical variables to numeric
data = pd.get_dummies(data, columns=['Sex', 'Embarked'], drop_first=True)

# Target and Features
X = data.drop('Survived', axis=1)
y = data['Survived']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

model = RandomForestClassifier(n_estimators=100, random_state=42)
model.fit(X_train, y_train)

y_pred = model.predict(X_test)

print("Accuracy:", accuracy_score(y_test, y_pred))
print("\nClassification Report:\n", classification_report(y_test, y_pred))
sns.heatmap(confusion_matrix(y_test, y_pred), annot=True, fmt='d', cmap='Blues')
plt.title('Confusion Matrix')
plt.show()

**Comment by vaishnavi1-lab** [comment]
import pandas as pd import numpy as np from sklearn.model_selection import
train_test_split, cross_val_score from sklearn.preprocessing import
StandardScaler from sklearn.linear_model import LogisticRegression from
sklearn.tree import DecisionTreeClassifier from sklearn.ensemble import
RandomForestClassifier from sklearn.svm import SVC from sklearn.metrics
import accuracy_score, precision_score, recall_score, f1_score,
confusion_matrix, classification_report import matplotlib.pyplot as plt
import seaborn as sns df = pd.read_csv('/content/iris.csv') print("First
few rows of the dataset:") print(df.head()) print("\nMissing values in the
dataset:") print(df.isnull().sum()) X = df.drop(columns=['species']) y = df[
'species'] X_train, X_test, y_train, y_test = train_test_split(X, y,
test_size=0.2, random_state=42, stratify=y) scaler = StandardScaler()
X_train = scaler.fit_transform(X_train) X_test = scaler.transform(X_test)
log_reg = LogisticRegression() decision_tree = DecisionTreeClassifier()
random_forest = RandomForestClassifier() svc = SVC() models = { "Logistic
Regression": log_reg, "Decision Tree": decision_tree, "Random Forest":
random_forest, "Support Vector Machine": svc } results = {} for name, model
in models.items(): model.fit(X_train, y_train) y_pred =
model.predict(X_test) accuracy = accuracy_score(y_test, y_pred) precision =
precision_score(y_test, y_pred, average='weighted') recall =
recall_score(y_test, y_pred, average='weighted') f1 = f1_score(y_test,
y_pred, average='weighted') results[name] = { "Accuracy": accuracy,
"Precision": precision, "Recall": recall, "F1 Score": f1 } print(f"\n{name}
- Classification Report:") print(classification_report(y_test, y_pred))
results_df = pd.DataFrame(results).T print("\nComparison of model
performances:") print(results_df) results_df.plot(kind='bar', figsize=(14, 8))
plt.title('Comparison of Classification Models') plt.xlabel('Model')
plt.ylabel('Score') plt.xticks(rotation=45) plt.legend(loc='lower right')
plt.grid(True) plt.show()

First few rows of the dataset:
   sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

Missing values in the dataset:
sepal_length    0
sepal_width     0
petal_length    0
petal_width     0
species         0
dtype: int64

Logistic Regression - Classification Report:
              precision    recall  f1-score   support

      setosa       1.00      1.00      1.00        10
  versicolor       0.90      0.90      0.90        10
   virginica       0.90      0.90      0.90        10

    accuracy                           0.93        30
   macro avg       0.93      0.93      0.93        30
weighted avg       0.93      0.93      0.93        30


Decision Tree - Classification Report:
              precision    recall  f1-score   support

      setosa       1.00      1.00      1.00        10
  versicolor       0.82      0.90      0.86        10
   virginica       0.89      0.80      0.84        10

    accuracy                           0.90        30
   macro avg       0.90      0.90      0.90        30
weighted avg       0.90      0.90      0.90        30


Random Forest - Classification Report:
              precision    recall  f1-score   support

      setosa       1.00      1.00      1.00        10
  versicolor       0.82      0.90      0.86        10
   virginica       0.89      0.80      0.84        10

    accuracy                           0.90        30
   macro avg       0.90      0.90      0.90        30
weighted avg       0.90      0.90      0.90        30


Support Vector Machine - Classification Report:
              precision    recall  f1-score   support

      setosa       1.00      1.00      1.00        10
  versicolor       1.00      0.90      0.95        10
   virginica       0.91      1.00      0.95        10

    accuracy                           0.97        30
   macro avg       0.97      0.97      0.97        30
weighted avg       0.97      0.97      0.97        30


Comparison of model performances:
                        Accuracy  Precision    Recall  F1 Score
Logistic Regression     0.933333   0.933333  0.933333  0.933333
Decision Tree           0.900000   0.902357  0.900000  0.899749
Random Forest           0.900000   0.902357  0.900000  0.899749
Support Vector Machine  0.966667   0.969697  0.966667  0.966583

[image: image.png]
